### DISCUSSION TITLE
How to specify examples of the responses?

### DISCUSSION THREAD

**Discussion by DimShadoWWW** [go-fuego/fuego#244] [General]
I tried to specify examples of responses on the endpoints but could not find how in the code.

All example values in OpenAPI are autogenerated and do and do not take into account the name of tags if the response is in xml format.

The example for this schema:
```
type Updates struct {
	XMLName xml.Name `xml:"updates" json:"-"`
	Channel []string `xml:"channel,omitempty" json:"channel,omitempty" validate:"omitempty,dive,http_url"`
}
```
is this one:
```
<?xml version="1.0" encoding="UTF-8"?>
<Updates>
	<channel>string</channel>
</Updates>
```

instead of:
```
<?xml version="1.0" encoding="UTF-8"?>
<updates>
	<channel>string</channel>
</updates>
```

**Comment by thunderstorm99** [comment]
I hope this will help you:
I specified a type with an example tag like so:
```go
type GenericAnswer struct {
	Message string `json:"message" example:"this is an example"`
}
```

after declaring my server with:
```go
mux := fuego.NewServer(fuego.WithAddr("127.0.0.1:8000"))
```

i register a Get endpoint with some options:
```go
fuego.Get(mux, "/ready", readiness, fuego.OptionAddResponse(http.StatusTeapot, "Teapot", fuego.Response{
	Type: GenericAnswer{},
}))
```

the used readiness function looks like this:
```go
func readiness(c fuego.ContextNoBody) (GenericAnswer, error) {
	if !ready {
		return GenericAnswer{Message: "not ready"}, fuego.HTTPError{Status: http.StatusTeapot}
	}

	return GenericAnswer{Message: "ready"}, nil
}
```
`ready` is just a global variable

running the server with
```go
mux.Run()
```

will yield the following swagger docs:
<img width="2529" height="1711" alt="Screenshot 2025-09-16 at 10-17-23 OpenAPI specification" src="https://github.com/user-attachments/assets/dbc2f981-deeb-46b3-9222-39119d0bd40c" />
